src/status_im2/contexts/chat/bottom_sheet_composer/utils.cljs

```diff
@@ -0,0 +1,72 @@
+(ns status-im2.contexts.chat.bottom-sheet-composer.utils
+  (:require
+    [oops.core :as oops]
+    [react-native.platform :as platform]
+    [react-native.reanimated :as reanimated]
+    [status-im2.contexts.chat.bottom-sheet-composer.constants :as c]))
+
+(defn bounded-val
+  [f min max]
+  (Math/max min (Math/min f max)))
```

Comment:
It seems to not work with functions. The first argument to `Math/min` is a function. Using `min` from clojure crashes the app. Also, the second argument to `Math/max` is still a function.

Reply:
Now that you mention, I don't really understand this `bounded-val` function. Calling `Math.min(() => 3, 2) ` in Javascript returns a `NaN` and I couldn't find in the Javascript documentation any support for passing a function to `Math.min` or `Math.max`.

Could you explain how this function works at all, please?

Reply:
@ilmotta `bounded-val` would execute a calculation and return the result. If the result is less than a `min` value, then return `min`. Likewise for `max`.

Idk why `Math.min(() => 3, 2)` return `Nan`.

`Math.min(2, Math.max(4,5))` works with no problem.

`(typeof Math.max)` returns `function`
`(typeof (() => 3))` returns `function`

Reply:
It seems that it can't except an arrow function. It can accept a reference to a function. The following works:

`function somFunc() {return 3}`
`Math.min(2, somFunc())` returns 2

Reply:
I don't see in the Javascript specs anything mentioning accepting functions, only numbers. Unless we find the documentation saying this is correct, I think we should change some names in this function, because it's quite confusing.

I traced all calls to `bounded-val`, and all of them pass numbers to `bounded-val`, never functions. So here's my suggestion for refactoring.

1. Avoid naming things as `min` and `max`, since these are `cljs.core` functions. Otherwise we'd be shadowing them.
2. Use the `min` and `max` core functions.

```clojure
(defn bounded-val
  [val min-val max-val]
  (max min-val (min val max-val)))

(bounded-val 10 9 11) ; => 10
(bounded-val 9 9 11) ; => 9
(bounded-val 8 9 11) ; => 9
(bounded-val 12 9 11) ; => 11
```

Reply:
It is working now. Thanks @ilmotta !